volume-providers/vpc/vpcclient/instances/list_volume_attachments.go

```diff
@@ -20,7 +21,10 @@ import (
 
 // ListVolumeAttachment retrives the list volume attachments with givne volume attachment details
 func (vs *VolumeAttachService) ListVolumeAttachment(volumeAttachmentTemplate *models.VolumeAttachment, ctxLogger *zap.Logger) (*models.VolumeAttachmentList, error) {
-	defer util.TimeTracker("GetAttachStatus", time.Now())
+
+	methodName := "VolumeAttachService.ListVolumeAttachment"
+	defer util.TimeTracker(methodName, time.Now())
```

Comment:
TimeTracker looks redundant. if its there everywhere...can we do the "metrics" logging from inside TimeTracker() ?  But if metrics scope is limited..and TimeTracker's usage scope is high..then we can leave it as it is.


defer util.TimeTracker(methodName, time.Now())
	defer metrics.UpdateDurationFromStart(ctxLogger, methodName, time.Now())

Adding a "methodName" variable can be avoided...unless until, thats used many places across the function. Better to use String Directly. Also a "metrics" naming convention can differ from local logging.

Reply:
methodName is used at multiple places in that method